integrationTests/vm/txsFee/validators_sc_test.go

```diff
@@ -0,0 +1,362 @@
+
+	recCode, err := testContextMeta.TxProcessor.ProcessTransaction(tx)
+	require.Equal(t, vmcommon.Ok, recCode)
+	require.Nil(t, err)
+	require.Nil(t, testContextMeta.GetLatestError())
+
+	intermediateTxs := testContextMeta.GetIntermediateTransactions(t)
+	scrWithMessage := intermediateTxs[1].(*smartContractResult.SmartContractResult)
+	// validator is in waiting queue
+	require.True(t, strings.Contains(string(scrWithMessage.ReturnMessage), "staking is full key put into waiting list "))
+
+	utils.CleanAccumulatedIntermediateTransactions(t, testContextMeta)
+
+	tx = vm.CreateTransaction(0, big.NewInt(0), sndAddr, vmAddr.ValidatorSCAddress, gasPrice, gasLimit, []byte("unStake@4fcdbfce9a3621621d388019353f87aceb0c5ec826256bc5a57220ed8f7d84b5c13f50219dc7a6ef090671f39d398c106f8a36022a04e6c18061ff38629d5b42b125aaea1e94a97b5f1c871bcf93b1f141a49ddb0b4c32b976cd530c1008da86"))
+
+	recCode, err = testContextMeta.TxProcessor.ProcessTransaction(tx)
+	require.Equal(t, vmcommon.Ok, recCode)
+	require.Nil(t, err)
+	require.Nil(t, testContextMeta.GetLatestError())
+
+	intermediateTxs = testContextMeta.GetIntermediateTransactions(t)
+	require.Equal(t, 2, len(intermediateTxs))
+
+	utils.CleanAccumulatedIntermediateTransactions(t, testContextMeta)
+
+	tx = vm.CreateTransaction(0, big.NewInt(0), sndAddr, vmAddr.ValidatorSCAddress, gasPrice, gasLimit, []byte("unBondTokens@"+hex.EncodeToString(stakeValue.Bytes())))
+
+	recCode, err = testContextMeta.TxProcessor.ProcessTransaction(tx)
+	require.Equal(t, vmcommon.UserError, recCode)
+	require.Nil(t, err)
+
+	intermediateTxs = testContextMeta.GetIntermediateTransactions(t)
+	require.Equal(t, 1, len(intermediateTxs))
+
+	scrWithMessage = intermediateTxs[0].(*smartContractResult.SmartContractResult)
+	require.True(t, strings.Contains(string(scrWithMessage.ReturnMessage), "cannot unBond tokens, the validator would remain without min deposit, nodes are still active"))
+}
+
+func TestValidatorsSC_DoStakeWithTopUpValueTryToUnStakeTokensAndUnBondTokens(t *testing.T) {
+	testContextMeta, err := vm.CreatePreparedTxProcessorWithVMsMultiShard(core.MetachainShardId, vm.ArgEnableEpoch{})
+
+	require.Nil(t, err)
+	defer testContextMeta.Close()
+
+	saveNodesConfig(t, testContextMeta, 1, 1, 1)
+
+	testContextMeta.BlockchainHook.(*hooks.BlockChainHookImpl).SetCurrentHeader(&block.MetaBlock{Nonce: 10000})
+
+	gasPrice := uint64(10)
+	gasLimit := uint64(4000)
+	sndAddr := []byte("12345678901234567890123456789012")
+	stakeValue, _ := big.NewInt(0).SetString("2700000000000000000000", 10)
```

Comment:
we can have vars like `value2500eGLD` for these (or 2700, 200 and so on)

Reply:
done.

Reply:
not quite done. those vars can be created once per package

Reply:
done.